src/walle-lib/map.cpp

```diff
@@ -0,0 +1,146 @@
+#include "map.hpp"
+#include <iostream>
+#include <ostream>
+#include <vector>
+#include <utility>
+
+
+Map::Map()  : tableau_map(taille, std::vector<int>(taille)){
+    taille = 4;
+    init_map();
+}
+
+Map::Map(int taille) : tableau_map(taille, std::vector<int>(taille)){
+    this -> taille = taille;
+    init_map();
+}
+
+void Map::init_map() {
+    for (int i = 0; i < taille; i++) {
+        for (int j = 0; j < taille; j++) {
+            tableau_map[i][j] = 0;
+        }
+    }
+}
+
+void Map::afficher_map() {
+    for (int i = 0; i < taille; i++) {
+        for (int j = 0; j < taille; j++) {
+            std::cout << tableau_map[i][j] << " ";
+        }
+        std::cout << std::endl;
+    }
+}
+
+void Map::ajouter_arbre(int nombre) {
+    for (int i = 0; i < nombre; i++) {
+        int x = rand() % taille;
+        int y = rand() % taille;
+        if (tableau_map[x][y] == 0) {
+            tableau_map[x][y] = 1;
```

Comment:
Pourquoi avoir choisi d'utiliser des entiers pour encoder les objets stocké dans le tableau. Utiliser une enumération (voir une enum class en c++) aurait-été un bien meilleur choix:
Voir: https://www.geeksforgeeks.org/enum-classes-in-c-and-their-advantage-over-enum-datatype/